kmelia/kmelia-war/src/main/java/org/silverpeas/components/kmelia/control/KmeliaSessionController.java

```diff
@@ -262,7 +264,18 @@ public static List<String> getLanguagesOfAttachments(ResourceReference resourceR
     return languages;
   }
 
-  private void init() {
+  public boolean isPasteNodeAllowed() throws ClipboardException {
+    SilverLogger.getLogger(this).info("isPasteNodeAllowed() nbSelectedObjects = {0}", getClipboardSelectedObjects().size());
+    return getClipboardSelectedObjects().stream().anyMatch(s->s.isDataFlavorSupported(NodeSelection.NodeDetailFlavor));
+  }
+
+  public boolean isPastePublicationAllowed(boolean isRoot) throws ClipboardException {
+    SilverLogger.getLogger(this).info("isPastePublicationAllowed() isRoot = {0} - nbSelectedObjects = {1}", isRoot,getClipboardSelectedObjects().size());
+    return getClipboardSelectedObjects().stream().anyMatch(s->s.isDataFlavorSupported(
+        PublicationSelection.PublicationDetailFlavor)) && (!isRoot || getNbPublicationsOnRoot()==0);
+  }
```

Comment:
The code 
```Java
(!isRoot || getNbPublicationsOnRoot()==0)
```
can be replaced by the already existing method:
```Java
isPublicationAllowed(isRoot)
```
after modifying the signature of the method to accept a boolean instead of the node and then in line 2340 we can then invoke it as following:
```Java
if (selection.isDataFlavorSupported(PublicationSelection.PublicationDetailFlavor) && isPublicationAllowed(targetNode.isRoot())) {
```